core/field_textinput.js

```diff
@@ -282,14 +287,17 @@ Blockly.FieldTextInput.prototype.widgetCreate_ = function() {
 Blockly.FieldTextInput.prototype.widgetDispose_ = function() {
   // Finalize value.
   this.isBeingEdited_ = false;
+  this.isTextValid_ = true;
   // No need to call setValue because if the widget is being closed the
   // latest input text has already been validated.
-  if (this.value_ !== this.text_) {
+  if (this.value_ != this.htmlInputValue_) {
```

Comment:
So I'm not a huge fan of this optimization. It adds a bunch of extra complexity to the field, just so we avoid calling a final re-render when we close the input (``widgetDispose``).
Without it, we have no need to maintain the last value in this.htmlInputValue_ and we would just always call a re-render.

In general, if code is not in a hot path, as in code that is executed most often (and this isn't since it only runs once when the user closes the editor), it's better to shoot for code readability and maintainability then to micro-optimize things. In the long run, it will reduce bugs and allow us to focus our energy optimizing the hot path instead which will make more users happy!